plugins/inputs/hystrix_stream/stream_parser.go

```diff
@@ -0,0 +1,183 @@
+	PropertyValueMetricsRollingStatisticalWindowInMilliseconds    int         `json:"propertyValue_metricsRollingStatisticalWindowInMilliseconds"`
+	PropertyValueRequestCacheEnabled                              bool        `json:"propertyValue_requestCacheEnabled"`
+	PropertyValueRequestLogEnabled                                bool        `json:"propertyValue_requestLogEnabled"`
+	ReportingHosts                                                int         `json:"reportingHosts"`
+	ThreadPool                                                    string      `json:"threadPool"`
+}
+
+var (
+	healthy       = false
+	scanner       *bufio.Scanner
+	cachedEntries []HystrixStreamEntry
+	reader        io.ReadCloser
+	cacheLock     sync.Mutex
+)
+
+func latestEntries(url string) ([]HystrixStreamEntry, error) {
+
+	if !healthy {
+		resp, err := http.Get(url)
+		if err != nil {
+			return make([]HystrixStreamEntry, 0), err
+		}
+		scanner = bufio.NewScanner(resp.Body)
+		reader = resp.Body
+		cachedEntries = make([]HystrixStreamEntry, 0)
+		go fillCacheForever(scanner)
+		healthy = true
+		log.Printf("I! Initialized hystrix-input with url : [%s]", url)
+	}
+
+	if scanner.Err() != nil {
+		log.Printf("E! Error scanning hystrix-servlet: [%v]", scanner.Err())
+		reader.Close()
+		healthy = false
+		return make([]HystrixStreamEntry, 0), scanner.Err()
+	}
+
+	defer clearCache()
+	return cachedEntries, nil
+}
+
+func clearCache() {
+	cacheLock.Lock()
+	cachedEntries = cachedEntries[:0]
+	cacheLock.Unlock()
+}
+
+func fillCacheForever(scanner *bufio.Scanner) {
+	fillCacheForeverMax(scanner, 100000)
+}
+
+func fillCacheForeverMax(scanner *bufio.Scanner, maxEntries int) {
+	newEntryCounter := 0
+	for scanner.Err() == nil {
+		chunks := streamToStrings(scanner)
+		for _, chunk := range chunks {
+			entries, err := parseChunk(chunk)
+			if err == nil {
+				for _, entry := range entries {
+					cacheLock.Lock()
+					cachedEntries = append(cachedEntries, entry)
+					newEntryCounter++
+					cacheLock.Unlock()
+				}
+			}
+		}
+		if newEntryCounter >= maxEntries {
+			return
+		}
+	}
+}
+
+func parseChunk(streamChunk string) ([]HystrixStreamEntry, error) {
+
+	entries := make([]HystrixStreamEntry, 0)
+	for _, line := range strings.Split(streamChunk, "\n") {
+		if strings.Contains(line, "data:") {
+			entryPartOfLine := strings.SplitAfter(line, "data:")
+			if len(entryPartOfLine) == 2 {
+				entry := HystrixStreamEntry{}
+				jsonErr := json.Unmarshal([]byte(entryPartOfLine[1]), &entry)
+				if jsonErr != nil {
+					return entries, jsonErr
+				} else if entry.Type == "HystrixCommand" {
+					entries = append(entries, entry)
+				}
+			}
+		}
+	}
+
+	return entries, nil
+}
+
+func streamToStrings(scanner *bufio.Scanner) []string {
+	result := make([]string, 0)
+	for scanner.Scan() {
+		text := scanner.Text()
+		if isData(text) {
+			result = append(result, scanner.Text())
+			break
```

Comment:
Is this function supposed to only return the first non-empty line?  If so why return a slice of strings?

Reply:
Thanks, dont know what I was thinking here